### PR TITLE
Remove unnecessary dependencies

### PR DESCRIPTION
**Pull request by WonyoungChoi**
- Remove unnecessary dependencies to reduce sync time and download size.  (save about 4GB)
- Following packages will be removed:
  - Android and iOS related packages.
  - Fuchsia related packages.
  - vulkan, angle

### CODE REVIEW

**Comment by bwikbs** [comment]
It's just my opinion, but I think it's better not to make a difference with the upstream if possible. 
It's a personal preference.. so It's not a strong claim :smile:  
BTW, is this because of ci?

**Comment by WonyoungChoi** [comment]
@bwikbs Yes, I'm testing moving engine's CI to GitHub Actions. I made incremental build possible using cache in the hosted runner environment (not self-hosted). The build could be shortened to approximately 10 minutes. However, the `gclient sync` was too large to cache, spending most of the time building. This PR can be saved 3~4 minutes more. 
This PR is not necessarily necessary because the time saved is not significant. I also want to hear people's opinions.

**Comment by WonyoungChoi** [comment]
CI Testing log can be found in https://github.com/flutter-tizen/engine/actions/runs/1149492619
The reasons for using github hosted runner instead of self-hosted runner are as follows:
- security risk : https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#self-hosted-runner-security-with-public-repositories
- concurrency : matrix jobs will be started simultaneously.

**Comment by bbrto21** [comment]
I think we can make a patch as a file and store some where such as tizen_tools and we can use whenever we want

**Comment by WonyoungChoi** [comment]
> I think we can make a patch as a file and store some where such as tizen_tools and we can use whenever we want

Yes that's good idea. I will try testing with a patch or using gclient commands more.

**Comment by bwikbs** [comment]
I'm looking forward to ``GitHub Actions`` stuff. I don't have any complaints now, but it's a desire to solve everything in one place...

**Comment by swift-kim** [comment]
> I think we can make a patch as a file and store some where such as tizen_tools and we can use whenever we want

Why? A commit is better than a patch.

> It's just my opinion, but I think it's better not to make a difference with the upstream if possible.

File: DEPS [hunk 1]
```diff
@@ -375,20 +378,20 @@ deps = {
...
+   # 'src/third_party/vulkan':
+   # Var('github_git') + '/KhronosGroup/Vulkan-Docs.git' + '@' + 'v1.1.91',
 
    'src/third_party/swiftshader':
    Var('swiftshader_git') + '/SwiftShader.git' + '@' + '5d1e8540407c138f47028d64684f3da599430aa4',
 
-   'src/third_party/angle':
-   Var('github_git') + '/google/angle.git' + '@' + '06d194e2ae7b1d7e0eda0c0c911eff92dec7d3d1',
+   # 'src/third_party/angle':
+   # Var('github_git') + '/google/angle.git' + '@' + '06d194e2ae7b1d7e0eda0c0c911eff92dec7d3d1',
```
Comment (swift-kim) [inline DEPS:388]: Can we exclude `swiftshader` as well?

File: DEPS [hunk 2]
```diff
@@ -550,7 +553,7 @@ deps = {
         'version': '0Db2pEb0Ur2hDgldKWY6Y07zH0a9po6LFu4gG9L9W1IC'
        }
      ],
-     'condition': 'host_os == "linux"',
+     'condition': 'download_linux_fuchsia_sdk',
```
Comment (swift-kim) [inline DEPS:556]: Why not just
```suggestion
     'condition': 'False',
```

**Review by swift-kim — COMMENTED** [review]
Overall this tradeoff seems reasonable for me (as a person who resolves conflicts on every upstream sync).

File: DEPS (hunk 1 above)
Comment (WonyoungChoi) [reply DEPS:388]: Yes we can exclude `swiftshader` in build job. But `test` job in current azurepipeline needs it. :(
https://github.com/flutter-tizen/engine/blob/2be5f48c2c200de23b2c4ffb12bd90b7b2d5a8da/azure-pipelines.yml#L20

**Comment by WonyoungChoi** [comment]
This changes will be replaced by using the script https://github.com/flutter-tizen/engine/blob/75f1e07887183204fccf519abe2a6e0176e6d1f1/ci/docker/tizen/prepare-sync.sh in the build workflow.

**Pull request closed without merging**